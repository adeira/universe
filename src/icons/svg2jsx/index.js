// @flow

// eslint-disable-next-line import/no-extraneous-dependencies
import svgr from '@svgr/core';

export default function transform(svgCode: string, iconName: string): Promise<string> {
  return svgr(
    svgCode,
    {
      icon: true, // Replace SVG "width" and "height" value by "1em" in order to make SVG size inherits from text size.
      plugins: ['@svgr/plugin-svgo', '@svgr/plugin-jsx', '@svgr/plugin-prettier'],
      template: ({ template }, opts, { componentName, jsx }) => {
        const code = `
          // @flow strict
          /* eslint-disable import/newline-after-import */

          import React, { type Node } from 'react';

          export default function %%COMPONENT_NAME%%(props: {}): Node {
            return %%JSX%%;
          };
        `;

        const flowTpl = template.smart(code, {
          plugins: ['flow'],
          preserveComments: true,
          syntacticPlaceholders: true,
        });

        return flowTpl({
          COMPONENT_NAME: componentName,
          JSX: jsx,
        });
      },
    },
    {
      componentName: iconName,
    },
  );
}
