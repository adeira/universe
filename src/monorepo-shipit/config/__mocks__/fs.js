// @flow

const fs: any = jest.createMockFromModule('fs');

fs.existsSync = (path) => path !== '/unknown_path';

export default fs;
