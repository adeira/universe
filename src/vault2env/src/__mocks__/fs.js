// @flow

const fs: any = jest.createMockFromModule('fs');

let mockFiles = Object.create(null);
fs.__setMockFiles = function (newMockFiles) {
  mockFiles = newMockFiles;
};

fs.existsSync = function (filePath) {
  return mockFiles[filePath] !== undefined;
};

fs.writeFileSync = function (filePath, content) {
  mockFiles[filePath] = content;
};

fs.readFileSync = function (filePath) {
  return mockFiles[filePath];
};

module.exports = fs;
