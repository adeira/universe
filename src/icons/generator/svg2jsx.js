// @flow

import fs from 'fs';
import path from 'path';
import { transform as svgrTransform } from '@svgr/core';

import prettify from './prettify';

export default function svg2jsx(svgCode: string, iconName: string): Promise<string> {
  return svgrTransform(
    svgCode,
    {
      icon: true, // replace SVG "width" and "height" value by "1em" in order to make SVG size inherits from text size
      svgo: true,
      svgoConfig: {
        multipass: true, // pass over SVGs multiple times to ensure all optimizations are applied
        plugins: [
          {
            name: 'preset-default',
            params: {
              overrides: {
                removeViewBox: false,
              },
            },
          },
        ],
      },
      plugins: ['@svgr/plugin-svgo', '@svgr/plugin-jsx'],
      // The latest SVGR templating is kinda dumb, and we really just want to get an optimized SVG.
      // So, instead of trying to create a custom template, we return a partial SVG and do the rest
      // later with our proper template (see below).
      template: (variables, { tpl }) => tpl`${variables.jsx}`,
    },
    {
      componentName: iconName,
    },
  )
    .then((partialJSX) => {
      return fs
        .readFileSync(path.join(__dirname, 'templates', 'reactComponent.txt'), 'utf8')
        .replace('%%JSX%%', partialJSX)
        .replace('%%COMPONENT_NAME%%', iconName);
    })
    .then(prettify);
}
