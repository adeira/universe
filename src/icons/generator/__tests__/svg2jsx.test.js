// @flow

import fs from 'fs';
import path from 'path';

import svg2jsx from '../svg2jsx';

it('works as expected', async () => {
  await expect(
    svg2jsx(
      fs.readFileSync(path.join(__dirname, '..', '..', 'original', 'backward.svg'), 'utf8'),
      'BackwardIcon',
    ),
  ).resolves.toMatchSnapshot();
});
