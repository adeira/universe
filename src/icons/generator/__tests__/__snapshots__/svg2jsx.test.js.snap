// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works as expected 1`] = `
"// @flow strict

/* eslint-disable import/newline-after-import */
import React, { type Node } from 'react';
export default function BackwardIcon(props: {}): Node {
  return (
    <svg height=\\"1em\\" viewBox=\\"0 0 21 21\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
      <path
        d=\\"M8.5 7.5c5.185-.47 8.52 1.53 10 6-2.825-3.14-6.341-3.718-10-2v3l-5-5 5-5z\\"
        fill=\\"none\\"
        stroke=\\"currentColor\\"
        strokeLinecap=\\"round\\"
        strokeLinejoin=\\"round\\"
      />
    </svg>
  );
}
"
`;
